texar/modules/networks/network_base.py

```diff
@@ -134,6 +134,12 @@ def layer_names(self) -> List[str]:
         """
         return self._layer_names
 
+    @property
+    def output_size(self) -> int:
+        r"""Return the finale dimension of :meth:`forward` output tensor.
```

Comment:
Again, I don't think it's necessary to implement an abstract method as an abstract method. Also, this module is not marked as ABC, which means it is a concrete class. Do `output_size` computations differ for all its derived classes?